enhancements/machine-api/converting-machine-api-to-cluster-api.md

```diff
@@ -0,0 +1,756 @@
+the migration is working, with the ability to minimise the impact to running clusters.
+
+#### Story 2
+
+As a cluster administrator, I would like to be able to verify the conversion of my Machine configuration, before instructing the cluster to operate
+on the changed configuration, so that I can check that the features I have tuned are converted correctly to the new configuration format.
+
+#### Story 3
+
+As a cluster administrator, I would like to test the new API before rolling it out to my production clusters, so that I can build confidence in the new APIs.
+To do this, I would like to be able to add Cluster API MachineSets to add new Machines to my Cluster without having to have an equivalent Machine API MachineSet to match it.
+
+#### Story 4
+
+As a developer of Machine related APIs, I would like to be able to gather feedback on Machine conversion layers before forcing new APIs onto users,
+so that we are confident when we go to GA the new APIs, that we will not break our existing users.
+
+### Goals
+
+* Allow users to migrate Machines between Machine API/Cluster API to test the new API
+* Allow users a choice (initially[^3]) of when they want to migrate
+* Eventually[^3] remove the Machine API controllers and leverage the migration layer to convert Machine API machines to Cluster API machines
+* Allow the Cluster Autoscaler, MachineHealthCheck and ControlPlaneMachineSet to continue to work during a semi-migrated cluster state
+* Ensure that every Machine API resource is translated to a Cluster API resource to allow users to test the new API
+* Allow access to newer features of Cluster API that do not exist in Machine API
+
+[^3]: In the first iteration of this project, the goal is to allow users to migrate individually resources.
+In later iterations, the plan is to remove the Machine API controllers. The linked goals will be executed in separate releases.
+
+### Non-Goals
+
+* Setting up or configuring any part of Cluster API, this is handled in a separate enhancement
+* Modifying any of the existing deployments of Machine API within clusters
+* Deleting or modifying any existing resources within clusters (until a user chooses to do so)
+* Force creation of new Machine API machines when new Cluster API MachineSets are being created[^4]
+* Port any feature from Cluster API into Machine API
+
+[^4]: We want to promote, over time, a reduction in the number of Machine API Machines, as such,
+any resources created directly in Cluster API, will not be reflected into Machine API, unless they have an owner,
+and, that owner is already reflected into Machine API.
+
+## Proposal
+
+We will implement a two-way synchronisation controller that synchronises Machines and related resources between the Machine API and Cluster API equivalents.
+
+Using an annotation on the Machine API resource, users will be able to choose which API is authoritative.
```

Comment:
Events are only created for create, update and delete events. What should happen is that the controller updates the CAPI resource, writes that, gets an event back for the update, then reconciles the CAPI machine, checks that the CAPI machine is up to date WRT the MAPI Machine, and then exit without writing an update. So I would expect for each write a second reconcile to stabilise. I don't think it will create an infinite loop if that's a concern